Bio/Application/__init__.py

```diff
@@ -383,7 +385,9 @@ def _check_value(self, value, name, check_function):
         """
         if check_function is not None:
             is_good = check_function(value)  # May raise an exception
-            assert is_good in [0, 1, True, False]
+            if is_good not in [0, 1, True, False]:
+                raise ValueError("Result of check_function: %s is of an unexpected value"
```

Comment:
Can you use ``%r`` for the Python object representation - that seems safer given we don't know what kind of value has been passed in.